apps/prairielearn/src/pages/administratorQuery/administratorQuery.html.ts

```diff
@@ -0,0 +1,353 @@
+import { html, unsafeHtml } from '@prairielearn/html';
+import { renderEjs } from '@prairielearn/html-ejs';
+import { z } from 'zod';
+
+import { nodeModulesAssetPath } from '../../lib/assets';
+
+export const AdministratorQueryResultSchema = z.object({
+  rows: z.array(z.record(z.any())),
+  rowCount: z.number().optional(),
+  columns: z.array(z.string()).optional(),
```

Comment:
When will these be `undefined`?

Reply:
They should not be. Sorry, had tried optional in debugging and missed removing them on commit. Removed in commit 08ab6e2.